google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -543,17 +553,32 @@ def update
         #   file.download "path/to/downloaded/gzipped.txt",
         #                 skip_decompress: true
         #
-        def download path = nil, verify: :md5, encryption_key: nil,
+        # @example Partially download.
+        #
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #   bucket = storage.bucket "my-bucket"
+        #   file = bucket.file "path/to/my-file.ext"
+        #
+        #   downloaded = file.download range: "bytes=3-6"
+        #   downloaded.rewind
+        #   downloaded.read #=> "lo w"
+        #
+        def download path = nil, verify: :md5, encryption_key: nil, range: nil,
                      skip_decompress: nil
           ensure_service!
           if path.nil?
             path = StringIO.new
             path.set_encoding "ASCII-8BIT"
           end
-          file, resp = service.download_file \
-            bucket, name, path, key: encryption_key, user_project: user_project
+          file, resp =
+            service.download_file bucket, name, path,
+                                  key: encryption_key, range: range,
+                                  user_project: user_project
           # FIX: downloading with encryption key will return nil
```

Comment:
@quartzmo could you provide more context on the `Fix` comment?

@georgeclaghorn I'd like to see this PR include a test for a ranged GET when the file is encrypted.

Reply:
@frankyn It was @blowmage who added that comment, in 5b6c0355, so he might be able to tell us.

Reply:
The same comment appears in [file_test.rb#L180](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/5b6c035540da45de4988d85ed34b98cd607963f3/google-cloud-storage/test/google/cloud/storage/file_test.rb#L180).

Reply:
Usually `service.download_file` returns the File/IO object the contents were written to, except that it returns `nil` when using an encryption key. So when the variable is `nil` we are setting it to a File object of the path that the contents were written to.